### PR TITLE
fix: SCORM attachment link error

### PR DESCRIPTION
**Pull request by edalex-yinzi**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change
Fix missing attachment links for unzip files.

The issue is in `staging` folder all unzip files are not transferred properly and still stay in the folder `_uploads` which will be removed before moving to `institution` folder.

#3465 

## Before:

Upload a zip file:
166291304931a5d3823c1514bed35ac5

Unzip:
f2202021b2a9a5f3d9751e12d428351b

Save:
02adc19cd9e4bde824e0b284016e1653

## After fix

Upload zip file:
image

Unzip:
image

Save:
image

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/controls/universal/handlers/FileUploadHandlerNew.scala [hunk 1]
```diff
@@ -288,8 +288,11 @@ class FileUploadHandlerNew extends AbstractAttachmentHandler[FileUploadHandlerMo
         editingArea.foreach(sf => stagingService.removeStagingArea(sf, true))
       def commit(): Unit = (a, zipAttachment) match {
         case (fa: FileAttachment, true) =>
-          stagingContext.moveFile(fa.getFilename, WebFileUploads.zipPath(fa.getFilename))
-          commitUnzipPaths.foreach(p => stagingContext.moveFile(p._1, fa.getFilename))
+          val UPLOADS_FOLDER = FileUploadState.UPLOADS_FOLDER
+          val newName        = fa.getFilename.replaceFirst(s"^$UPLOADS_FOLDER/.*/", "")
+          stagingContext.moveFile(fa.getFilename, WebFileUploads.zipPath(newName))
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/controls/universal/handlers/FileUploadHandlerNew.scala:293]: Can you add some comments here to describe why you do this ?

**Comment by edalex-ian** [comment]
I imagine two things:

1. This used to work, and hence why it was raised as a bug only recently; and
2. This probably came about as a result of the rework of the upload control.

With that in mind, perhaps we should look back to how it used to be done, starting with the following references to when the change was made:

- This issue: #267 (which unfortunately lacks and 'why' for such a large overhaul - may have been the desire to move more code to scala and purescript)
- This PR: #291
- This final commit into master: 20ee6ceb6a55d5197ded4d49f41332033ead33fd

Keep in mind that since those the purscript code was replaced with typescript and the scala updated as needed.

**Comment by edalex-yinzi** [comment]
I found this bug is first mentioned in 2018 but I was confused since it says it was fixed. https://github.com/openequella/openEQUELLA/issues/576

**Comment by edalex-yinzi** [comment]
Could it be because that change was not merged to develop?  I can't find that commit history in develop branch locally.

**Comment by PenghaiZhang** [comment]
> Could it be because that change was not merged to develop? I can't find that commit history in develop branch locally.

If you open any file involved in that commit, can you see the changes ?

**Comment by PenghaiZhang** [comment]
> Could it be because that change was not merged to develop? I can't find that commit history in develop branch locally.

You can also try to cherry pick that commit in your working branch and see if it can fix the issue.

**Comment by PenghaiZhang** [comment]
That commit is not a merge commit. I think it was pushed to stable-6.6 directly.

**Comment by edalex-ian** [comment]
Work on this continues elsewhere, closing PR.

**Pull request closed without merging**